webapp/channels/src/components/commercial_support_modal/commercial_support_modal.tsx

```diff
@@ -61,6 +69,50 @@ export default class CommercialSupportModal extends React.PureComponent<Props, S
         this.updateBannerWarning(false);
     };
 
+    updateCheckStatus = (index: number) => {
+        this.setState({
+            packetContents: this.state.packetContents.map((content, currentIndex) => (
+                (currentIndex === index && !content.mandatory) ? {...content, selected: !content.selected} : content
+            )),
+        });
+    };
+
+    genereateDownloadURLWithParams = (): string => {
+        const url = new URL(Client4.getSystemRoute() + '/support_packet');
+        this.state.packetContents.forEach((content) => {
+            if (content.id === 'basic.server.logs') {
+                url.searchParams.set('basic.server.logs', String(content.selected));
+            } else if (!content.mandatory && content.selected) {
+                url.searchParams.append('plugin_packets', content.id);
+            }
+        });
+        return url.toString();
+    };
+
+    downloadSupportPacket = () => {
+        this.setState({loading: true});
+        fetch(this.genereateDownloadURLWithParams(), {
+            method: 'GET',
+            headers: {'Content-Type': 'application/zip'},
+        }).then((res) => {
+            return res.blob();
+        }).finally(() => {
+            this.setState({loading: false});
+        }).then((blob) => {
+            // we emulate the server filenaming convention to maintain file name
+            const formattedDate = (moment(new Date())).format('YYYY-MM-DD-HH-mm');
+            const href = window.URL.createObjectURL(blob);
+            const link = document.createElement('a');
+            link.href = href;
+            link.setAttribute('download', `mattermost_support_packet_${formattedDate}.zip`);
+            document.body.appendChild(link);
+            link.click();
+            document.body.removeChild(link);
+        }).catch((err) => {
+            return Promise.reject(err);
+        });
+    };
```

Comment:
I think we can simplify this by doing something like: 
```
    downloadSupportPacket = async () => {
        this.setState({loading: true});
        const res = await fetch(this.genereateDownloadURLWithParams(), {
            method: 'GET',
            headers: {'Content-Type': 'application/zip'},
        });
        const blob = await res.blob();
        this.setState({loading: false});

        // we emulate the server filenaming convention to maintain file name
        const formattedDate = (moment(new Date())).format('YYYY-MM-DD-HH-mm');
        const href = window.URL.createObjectURL(blob);
        const link = document.createElement('a');
        link.href = href;
        link.setAttribute('download', `mattermost_support_packet_${formattedDate}.zip`);
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
    };
```

Reply:
But wow, I'm surprised this is the best way to do. Seems hacky. 😅

Reply:
It's much better with `async` now :_)